validator/length.go

```diff
@@ -0,0 +1,127 @@
+package validator
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/cobra"
+)
+
+// Length is a struct that provides a map
+// with key as attribute for which length is controlled
+// and value limit as Limit struct
+type Length struct {
+	Limits map[string]Limit
+}
+
+// Limit defines min, max length of string
+type Limit struct {
+	Min, Max int
+}
+
+// Validate implements the Rule interface
+func (l *Length) Validate(cmd *cobra.Command, verbose bool) []Error {
+	var errors []Error
+	info := stats{num: 0, num_failed: 0, errors: errors}
+
+	return l.ValidateHelper(cmd, verbose, info)
+}
+
+// ValidateHelper returns errors of type []Error
+// if length is not validated
+func (l *Length) ValidateHelper(cmd *cobra.Command, verbose bool, info stats) []Error {
+
+	// validate the root command
+	err := validateLength(cmd, l, verbose)
+	// record stats
+	info.num++
+	info.num_failed += len(err)
+	info.errors = append(info.errors, err...)
```

Comment:
We need errors to have more than string - struct with cobra command and rule name that caused it

Reply:
yes i did it
```go
type Error struct {
	Name string
	Err  error
	Rule string
}
```

Reply:
Too generic name. Lets change it